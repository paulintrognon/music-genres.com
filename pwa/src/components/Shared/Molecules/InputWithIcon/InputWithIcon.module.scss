.container {
  position: relative;
  display: flex;
  align-items: center;
}

.icon {
  position: absolute;
  width: auto;
  right: 1rem;
}
