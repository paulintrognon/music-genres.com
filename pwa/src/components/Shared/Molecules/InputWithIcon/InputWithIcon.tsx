import TextInput, { Props as TextInputProps } from '../../Atoms/TextInput/TextInput'
import classnames from 'classnames'
import classes from './InputWithIcon.module.scss'

interface Props extends TextInputProps {
  icon: React.ReactNode
  onSubmit?: () => void
}
const InputWithIcon: React.FC<Props> = ({ icon, className, onSubmit, ...props }) => {
  const handleOnSubmit = (event: React.FormEvent): void => {
    event.preventDefault()
    onSubmit && onSubmit()
  }

  return (
    <form className={classnames(classes.container, className)} onSubmit={handleOnSubmit}>
      <TextInput {...props} />
      <button type="submit" className={classes.icon}>
        {icon}
      </button>
    </form>
  )
}
export default InputWithIcon
