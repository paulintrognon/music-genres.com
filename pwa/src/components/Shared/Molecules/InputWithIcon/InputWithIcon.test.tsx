import InputWithIcon from './InputWithIcon'
import { shallow } from 'enzyme'

it('should render the InputWithIcon', () => {
  const wrapper = shallow(<InputWithIcon icon={<span className="the-icon" />} />)
  expect(wrapper).toMatchSnapshot()
})

it('should call onSubmit on form submit', () => {
  const onSubmit = jest.fn()
  const wrapper = shallow(<InputWithIcon icon={null} onSubmit={onSubmit} />)
  wrapper.find('form').first().simulate('submit', { preventDefault: jest.fn() })

  expect(onSubmit).toBeCalledTimes(1)
})
