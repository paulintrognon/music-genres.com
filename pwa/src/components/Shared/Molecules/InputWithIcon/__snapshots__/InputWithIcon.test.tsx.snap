// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the InputWithIcon 1`] = `
<form
  className="container"
  onSubmit={[Function]}
>
  <TextInput />
  <button
    className="icon"
    type="submit"
  >
    <span
      className="the-icon"
    />
  </button>
</form>
`;
